barrels/LogMonkey/Logger.mc

```diff
@@ -0,0 +1,172 @@
+                currentTime.min.format("%02u"),
+                currentTime.sec.format("%02u")
+            ]);
+
+            // Form the log message
+            return Lang.format(FORMAT_LOG_MESSAGE, [timestamp, mLogLevel, tag, message]);
+        }
+
+        (:release)
+        private function formLogMessage(tag, message) {
+        }
+
+        //! Handles the printing of a log message to the log file
+        //! @param [Toybox::Lang::String] tag The tag to apply to the log message
+        //! @param [Toybox::Lang::String] message The message to log
+        (:debug)
+        private function log(tag, message) {
+            // Print the log message
+            mLogStream.println(formLogMessage(tag, message));
+        }
+
+        (:release)
+        private function log(tag, message) {
+        }
+
+        //! Returns the type string of the given variable. Note, this function
+        //! will currently only return names within the Toybox.Lang module. If the
+        //! given variable isn't a type defined there then "Object" will be returned.
+        //! @param [Toybox::Lang::Object] variable The variable to get the type of
+        //! @return [Toybox::Lang::String] The name of the type of the given variable
+        (:debug)
+        private function getVariableType(variable) {
+            // If the given value is null we can't switch on it so perform
+            // a null check here. The return value should just be "null".
+            if (variable == null) {
+                return "null";
+            }
+
+            // Switch on the type of the variable and return the variable's
+            // class name accordingly.
+            switch (variable) {
+                case instanceof Lang.Array:
+                    return "Array";
+                case instanceof Lang.Boolean:
+                    return "Boolean";
+                case instanceof Lang.Char:
+                    return "Char";
+                case instanceof Lang.Dictionary:
+                    return "Dictionary";
+                case instanceof Lang.Double:
+                    return "Double";
+                case instanceof Lang.Exception:
+                    return "Exception";
+                case instanceof Lang.Float:
+                    return "Float";
+                case instanceof Lang.Long:
+                    return "Long";
+                case instanceof Lang.Method:
+                    return "Method";
+                case instanceof Lang.Number:
+                    return "Number";
+                case instanceof Lang.String:
+                    return "String";
+                case instanceof Lang.Symbol:
+                    return "Symbol";
+                case instanceof Lang.WeakReference:
+                    return "WeakReference";
+                default:
+                    return "Object";
+            }
+        }
+
+        (:release)
+        private function getVariableType(variable) {
```

Comment:
This method is not needed in release. It should probably be omitted.

Reply:
Done